### PR TITLE
Prepare docs for 3.5

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Currently is just the merging of What's new. Need to work on the API changes next.

## PR Checklist

- [n/a] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/users/prev_whats_new/whats_new_3.5.0.rst [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+Figure now has ``draw_no_output`` method
+----------------------------------------
+
+Rarely, the user will want to trigger a draw without making output to either
+the screen or a file. This is useful for determining the final position of
+artists on the figure that require a draw, like text artists. This could be
+accomplished via ``fig.canvas.draw()`` but has side effects, sometimes requires
+an open file, and is documented on an object most users do not need to access.
+The `.Figure.draw_no_output` is provided to trigger a draw without pushing to
+the final output, and with fewer side effects.
```
Comment (story645) [inline doc/users/prev_whats_new/whats_new_3.5.0.rst:26]: This paragraph is very hard to follow if you don't already know what draw means, so I'm wondering if it's linking to some general documentation about draw? Or if there's a one sentence summary of draw? Or using the classic example of `ax.get_labels()` returns blank before draw?

```suggestion
Rarely, the user may want to trigger a draw without rendering output to either
the screen or a file. This is useful for determining the position of
artists that require a draw, like text artists. This could be
accomplished via ``fig.canvas.draw()`` but has side effects, sometimes requires
an open file, and is documented on an object most users do not need to access.
The `.Figure.draw_no_output` is provided to trigger a draw without rendering and with fewer side effects.
```

**Comment by QuLogic** [comment]
Oops, meant to be Draft.

File: doc/api/prev_api_changes/api_changes_3.5.0/behaviour.rst [hunk 2]
```diff
@@ -0,0 +1,246 @@
+Behaviour changes
+-----------------
+
+First argument to ``subplot_mosaic`` renamed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Both `.FigureBase.subplot_mosaic`, and `.pyplot.subplot_mosaic` have had the
+first positional argument renamed from *layout* to *mosaic*. As we are
+considering consolidating the *constrained_layout* and *tight_layout* keyword
+arguments in the Figure creation functions of `.pyplot` into a single *layout*
```
Comment (jklymak) [inline doc/api/prev_api_changes/api_changes_3.5.0/behaviour.rst:10]:
```suggestion
first positional argument renamed from *layout* to *mosaic*. As we have 
consolidated the *constrained_layout* and *tight_layout* keyword
arguments in the Figure creation functions of `.pyplot` into a single *layout*
```

File: doc/users/prev_whats_new/whats_new_3.5.0.rst [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+Images in Matplotlib created via `~.axes.Axes.imshow` are resampled to match
+the resolution of the current canvas. It is useful to apply an anto-aliasing
+filter when downsampling to reduce Moiré effects. By default, interpolation is
+done on the data, a norm applied, and then the colormapping performed.
+
+However, it is often desireable for the anti-aliasing interpolation to happen
+in RGBA space, where the colors are interpolated rather than the data. This
+usually leads to colors outside the colormap, but visually blends adjacent
+colors, and is what browsers and other image processing software does.
```
Comment (jklymak) [inline doc/users/prev_whats_new/whats_new_3.5.0.rst:126]:
```suggestion
colors, and is what browsers and other image processing software do.
```

**Review by jklymak — COMMENTED** [review]
What a huge amount of work!  Thanks @QuLogic.  

I found a couple of typos in my quick skim.

It seems we (I) have not properly outlined the changes in colorbar.  Not sure if that should be an API change or a whats-new, but we definitely need a note in here that colorbars have been majorly reworked so that the axes are more similar to normal axes.  Let me know where I should put it....

**Comment by QuLogic** [comment]
I think you can make an entry in both What's new, and the behaviour API notes.

File: doc/api/prev_api_changes/api_changes_3.5.0/removals.rst [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+
+Support for the (undocumented) ``TTFPATH`` and ``AFMPATH`` environment
+variables has been removed. Register additional fonts using
+``matplotlib.font_manager.fontManager.addfont()``.
+
+Modules
+~~~~~~~
+
+- ``matplotlib.backends.qt_editor.formsubplottool``; use
+  ``matplotlib.backends.backend_qt5.SubplotToolQt`` instead.
```
Comment (tacaswell) [inline doc/api/prev_api_changes/api_changes_3.5.0/removals.rst:123]:
```suggestion
  ``matplotlib.backends.backend_qt.SubplotToolQt`` instead.
```

This got moved around as part of the qt6 work.

**Review by tacaswell — APPROVED** [review]
